### ISSUE TITLE
[isolatedDeclarations] incorrect error when exported type is locally shadowed by an unexported variable

### ISSUE THREAD

**Issue by bradzacher** [oxc-project/oxc#12465]
# Repro

```ts
export type Foo = {};

export type Bar = {
    foo: Foo,
};

const Foo = new Map();
```
[playground](https://playground.oxc.rs/#eNpNULFOxDAM/ZUqE0gdAImliAWhbggEaxc3dU+RUjty3Lurqv47SUvLZUmc9/yen2djTWXwGli00ClgUTMXr8W8vDTU0C3wBpKBhop0euYqU8uGNqZlivrXTHgpPiDc3SfElIZNNRsZKV9xIoWrqVRGLI13pKbqwcdURMsBdyROQ8t+r1SAYs8ybB9LaQJIRMmK4D1fvlFHoc9Ro+uwHsmq4+S3dQfBxD3jFwhSPCSy+SqR3ofBpqkgJ0yTGYxPD4/PKYOL7EGxe0frQSDL/ytZ7vCEaz4kaD3+8CgWBwgHZXDkerfbpWWpsK/T5LnpjNJyTNnXTSzL8gvpqY/U)

# Expected

No error, [just like TS](https://www.typescriptlang.org/play/?isolatedDeclarations=true#code/KYDwDg9gTgLgBDAnmYcBiEJwLxwN4C+A3AFAmiSwLKoBCAhlDviXG3AGaYBc6mANCWJkAxhAB2AZ3gYsuccADucALL0wACgCUpIA)

# Actual

```
  x TS9010: Variable must have an explicit type annotation with
  | --isolatedDeclarations.
   ,-[:7:7]
 6 | 
 7 | const Foo = new Map();
   :       ^^^
   `----
```
